apis/v1alpha1/opentelemetrycollector_types.go

```diff
@@ -213,6 +213,22 @@ type OpenTelemetryCollectorSpec struct {
 	// +optional
 	InitContainers []v1.Container `json:"initContainers,omitempty"`
 
+	// AdditionalContainers allows injecting additional containers into the Collector's pod definition.
+	// These sidecar containers can be used for authentication proxies, logging shipping sidecars, agents for shipping
+	// metrics to their cloud, or in general sidecars that do not support automatic injection. This option only
+	// applies to Deployment, DaemonSet, and StatefulSet deployments of the collector. It does not apply to the sidecar
```

Comment:
deployment modes

Reply:
I think we should add validation to the validating webhook and return an error to user if the mode is sidecar and there are defined additional containers